src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -73,10 +73,44 @@ public OnnxNodeInfo(string name, OnnxShape shape, System.Type type)
         public readonly List<string> InputNames;
         public readonly List<string> OutputNames;
 
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
         public OnnxModel(string modelFile)
+            : this (modelFile, -1, false)
+        {
+        }
+
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
+        /// <param name="gpuDeviceId">Gpu device ID to execute on. -1 for CPU</param>
+        /// <param name="fallbackToCpu">If true, resumes CPU execution quitely upon GPU error</param>
+        public OnnxModel(string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false)
         {
             _modelFile = modelFile;
-            _session = new InferenceSession(modelFile);
+
+            if (gpuDeviceId >= 0)
+            {
+                try
+                {
+                    _session = new InferenceSession(modelFile, SessionOptions.MakeSessionOptionWithCudaProvider(gpuDeviceId));
+                }
+                catch (OnnxRuntimeException e)
```

Comment:
>OnnxRuntimeException [](start = 23, length = 20)

Are we guaranteed to always get an OnnxRuntimeException thrown when the InferenceSession fails?

Reply:
Yes. There will always be an OnnxRuntimeException upon any failure.